server/src/test/java/org/elasticsearch/index/IndexModuleTests.java

```diff
@@ -244,6 +246,39 @@ public void testRegisterIndexStore() throws IOException {
         indexService.close("simon says", false);
     }
 
+    public void testDirectoryWrapper() throws IOException {
+        final Path homeDir = createTempDir();
+        final Settings settings = Settings.builder()
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), homeDir.toString())
+            .build();
+        final IndexSettings indexSettings = IndexSettingsModule.newIndexSettings(index, settings);
+        final IndexModule module = new IndexModule(
+            indexSettings,
+            emptyAnalysisRegistry,
+            new InternalEngineFactory(),
+            Map.of(),
+            () -> true,
+            indexNameExpressionResolver,
+            Collections.emptyMap()
+        );
+
+        module.setDirectoryWrapper(new TestDirectoryWrapper());
+
+        final IndexService indexService = newIndexService(module);
+        assertSame(indexService.getEngineFactory(), module.getEngineFactory());
+        final IndexStorePlugin.DirectoryFactory directoryFactory = indexService.getDirectoryFactory();
+        assertThat(directoryFactory, notNullValue());
```

Comment:
Would it make sense to `assertTrue(indexService.getReaderWrapper() instanceof TestDirectoryWrapper);`?

Reply:
It's already covered by another test and we did not change that logic , so I'm not checking this again here.